test/unittest/dds/xtypes/dyndata_json/types/comprehensive_type/common/ComprehensiveCommon.cpp

```diff
@@ -0,0 +1,232 @@
+    dyn_data->return_loaned_value(dyn_data_short_long_map);
+
+    // union
+    traits<DynamicData>::ref_type dyn_data_inner_union = dyn_data->loan_value(dyn_data->get_member_id_by_name("inner_union"));
+    dyn_data_inner_union->set_int64_value(dyn_data_inner_union->get_member_id_by_name("second"), static_cast<int64_t>(index));
+    traits<DynamicData>::ref_type dyn_data_complex_union = dyn_data->loan_value(dyn_data->get_member_id_by_name("complex_union"));
+    dyn_data_complex_union->set_complex_value(dyn_data_complex_union->get_member_id_by_name("fourth"), dyn_data_inner_union);
+    dyn_data->return_loaned_value(dyn_data_complex_union);
+    dyn_data->return_loaned_value(dyn_data_inner_union);
+
+    // bitset
+    traits<DynamicData>::ref_type dyn_data_my_bitset = dyn_data->loan_value(dyn_data->get_member_id_by_name("my_bitset"));
+    dyn_data_my_bitset->set_int16_value(dyn_data_my_bitset->get_member_id_by_name("d"), static_cast<int16_t>(index));
+    dyn_data->return_loaned_value(dyn_data_my_bitset);
+}
+
+
+traits<DynamicData>::ref_type fill_dyn_data(
+        traits<DynamicData>::ref_type dyn_data,
+        const unsigned int& index)
+{
+    // index
+    dyn_data->set_uint32_value(dyn_data->get_member_id_by_name("index"), index);
+
+    // inner struct
+    traits<DynamicData>::ref_type dyn_data_all_struct = dyn_data->loan_value(dyn_data->get_member_id_by_name("inner_struct"));
+    fill_all_struct(dyn_data_all_struct, index);
+
+    // complex sequence
+    traits<DynamicData>::ref_type dyn_data_complex_sequence = dyn_data->loan_value(dyn_data->get_member_id_by_name("complex_sequence"));
+    dyn_data_complex_sequence->set_complex_value(0, dyn_data_all_struct);
+    dyn_data->return_loaned_value(dyn_data_complex_sequence);
+
+    // complex array
+    traits<DynamicData>::ref_type dyn_data_complex_array = dyn_data->loan_value(dyn_data->get_member_id_by_name("complex_array"));
+    for (unsigned int i = 0; i < 2; i++)
+    {
+        dyn_data_complex_array->set_complex_value(i, dyn_data_all_struct);
+    }
+    dyn_data->return_loaned_value(dyn_data_complex_array);
+
+    // complex map
+    traits<DynamicData>::ref_type dyn_data_complex_map = dyn_data->loan_value(dyn_data->get_member_id_by_name("complex_map"));
+    int16_t key {0};
+    dyn_data_complex_map->set_int16_value(dyn_data_complex_map->get_member_id_by_name(std::to_string(key)), static_cast<uint16_t>(index));
+    dyn_data_complex_map->set_complex_value(dyn_data_complex_map->get_member_id_by_name(std::to_string(1)), dyn_data_all_struct);
+    dyn_data->return_loaned_value(dyn_data_complex_map);
+
+    // Return AllStruct loaned DynamicData
+    dyn_data->return_loaned_value(dyn_data_all_struct);
+
+    return dyn_data;
+
+}
+
+template <>
+traits<DynamicType>::ref_type create_dynamic_type<DataTypeKind::COMPREHENSIVE_TYPE>()
+{
+    const auto type_name = "ComprehensiveType";
```

Comment:
I think you can get this from the TypeSupport.

Reply:
I did not realize you actually would need to instantiate a type support just to get the name. I don't like it, but I guess it's more correct than just hard code it.